### PR TITLE
[web] Fix `scale`

### PR DESCRIPTION
**Pull request by m-bert**
## Description

This PR does 2 things:

1. Adds check if `style.scale` is defined in order to avoid crash when trying to call `split` method
2. Fixes incorrect `scaleY` if only one value was present in `style.scale`

>[!NOTE]
>While I'm not sure why `style.scale` was `undefined` when it should be `none`, I suspect that this may be caused by `NativeWind`

## Test plan

Using code below I've verified that scale is calculated correctly by manipulating `element.style` in `inspector`.

<details>
<summary>Test code</summary>

```tsx
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Gesture, GestureDetector } from 'react-native-gesture-handler';

export default function EmptyExample() {
  const g = Gesture.Pan();
  return (
    <View style={styles.container}>
      <GestureDetector gesture={g}>
        <View style={styles.box} />
      </GestureDetector>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },

  box: {
    width: 100,
    height: 100,
    backgroundColor: 'plum',
    borderRadius: 10,
  },
});

```

</details>

Style            |  Logs
:-------------------------:|:-------------------------:
image  |   image

### CODE REVIEW

File: src/web/utils.ts [hunk 1]
```diff
@@ -27,28 +27,30 @@ export function calculateViewScale(view: HTMLElement) {
...
 
-  const scales = styles.scale.split(' ');
+  // Get scales from scale property
+  if (styles.scale !== undefined && styles.scale !== 'none') {
+    const scales = styles.scale.split(' ');
 
-  if (scales[0] !== 'none') {
-    resultScales.scaleX = parseFloat(scales[0]);
-  }
+    if (scales[0]) {
```
[the remark this answers is not in the document]
Comment (m-bert) [reply src/web/utils.ts:34]: `scales` is result of `split`, therefore elements inside this array are `string`. `'0'` won't be treated as `false` in that case 😅

File: src/web/utils.ts [hunk 2]
```diff
@@ -27,28 +27,30 @@ export function calculateViewScale(view: HTMLElement) {
...
+      resultScales.scaleX = parseFloat(scales[0]);
+    }
 
-  if (scales[1]) {
-    resultScales.scaleY = parseFloat(scales[1]);
+    resultScales.scaleY = scales[1]
```
[the remark this answers is not in the document]
Comment (m-bert) [reply src/web/utils.ts:38]: Same as above

File: src/web/utils.ts [hunk 3]
```diff
@@ -27,28 +27,30 @@ export function calculateViewScale(view: HTMLElement) {
     scaleY: 1,
   };
 
-  const scales = styles.scale.split(' ');
+  // Get scales from scale property
+  if (styles.scale !== undefined && styles.scale !== 'none') {
```
Comment (hirbod) [inline src/web/utils.ts:31]:
```suggestion
  if (typeof styles.scale !== undefined && styles.scale !== 'none') {
```

**Review by latekvo — APPROVED** [review]

**Review by j-piasecki — APPROVED** [review]

**Comment by EvertEt** [comment]
Hey @m-bert ! We came across an uptick of this split crash on our side as well. Do you know if there is a release planned soon including this fix?

Many thanks in advance and for this fix!

**Comment by j-piasecki** [comment]
Hi @EvertEt! Most likely this or the next week.

**Comment by EvertEt** [comment]
Thank you! 🤞

**Pull request merged**